sw/device/tests/sim_dv/spi_passthrough_test.c

```diff
@@ -0,0 +1,419 @@
+        .peripheral = kTopEarlgreyPinmuxOutselSpiHost1Sck,
+    },
+    {
+        .pad = kTopEarlgreyPinmuxMioOutIob1,
+        .peripheral = kTopEarlgreyPinmuxOutselSpiHost1Sd0,
+    },
+    {
+        .pad = kTopEarlgreyPinmuxMioOutIob3,
+        .peripheral = kTopEarlgreyPinmuxOutselSpiHost1Sd1,
+    },
+    // These peripheral I/Os are not assigned for tests.
+    //     {
+    //         .pad = ???,
+    //         .peripheral = kTopEarlgreyPinmuxOutselSpiHost1Sd2,
+    //     },
+    //     {
+    //         .pad = ???,
+    //         .peripheral = kTopEarlgreyPinmuxOutselSpiHost1Sd3,
+    //     },
+};
+
+static pinmux_select_t pinmux_in_config[] = {
+    {
+        .pad = kTopEarlgreyPinmuxInselIob1,
+        .peripheral = kTopEarlgreyPinmuxOutselSpiHost1Sd0,
+    },
+    {
+        .pad = kTopEarlgreyPinmuxInselIob3,
+        .peripheral = kTopEarlgreyPinmuxOutselSpiHost1Sd1,
+    },
+    // These peripheral I/Os are not assigned for tests.
+    //     {
+    //         .pad = ???,
+    //         .peripheral = kTopEarlgreyPinmuxOutselSpiHost1Sd2,
+    //     },
+    //     {
+    //         .pad = ???,
+    //         .peripheral = kTopEarlgreyPinmuxOutselSpiHost1Sd3,
+    //     },
+};
+
+enum spi_flash_opcode {
+  kSpiFlashOpReadJedec = 0x9f,
+  kSpiFlashOpReadSfdp = 0x5a,
+  kSpiFlashOpReadNormal = 0x03,
+  kSpiFlashOpReadFast = 0x0b,
+  kSpiFlashOpReadDual = 0x3b,
+  kSpiFlashOpReadQuad = 0x6b,
+  kSpiFlashOpWriteEnable = 0x06,
+  kSpiFlashOpWriteDisable = 0x04,
+  kSpiFlashOpReadStatus1 = 0x05,
+  kSpiFlashOpReadStatus2 = 0x35,
+  kSpiFlashOpReadStatus3 = 0x15,
+  kSpiFlashOpWriteStatus1 = 0x01,
+  kSpiFlashOpWriteStatus2 = 0x31,
+  kSpiFlashOpWriteStatus3 = 0x11,
+  kSpiFlashOpChipErase = 0xc7,
+  kSpiFlashOpSectorErase = 0x20,
+  kSpiFlashOpPageProgram = 0x02,
+  kSpiFlashOpEnter4bAddr = 0xb7,
+  kSpiFlashOpExit4bAddr = 0xe9,
+};
+
+enum spi_device_mem_offset {
+  kSpiDeviceMemOffsetSfdp = 0xc00,
+  kSpiDeviceMemOffsetPayload = 0xd00,
+};
+
+enum spi_device_command_slot {
+  kSpiDeviceReadCommandSlotBase = 0,
+  kSpiDeviceWriteCommandSlotBase = 11,
+};
+
+/**
+ * Initialize the provided SPI host. For the most part, the values provided are
+ * filler, as spi_host0 will be in passthrough mode and spi_host1 will remain
+ * unused throughout the test.
+ */
+void init_spi_host(dif_spi_host_t *spi_host,
+                   uint32_t peripheral_clock_freq_hz) {
+  dif_spi_host_config_t config = {
+      .spi_clock = peripheral_clock_freq_hz / 2,
+      .peripheral_clock_freq_hz = peripheral_clock_freq_hz,
+      .chip_select =
+          {
+              .idle = 2,
+              .trail = 2,
+              .lead = 2,
+          },
+  };
+  CHECK_DIF_OK(dif_spi_host_configure(spi_host, config));
+  CHECK_DIF_OK(dif_spi_host_output_set_enabled(spi_host, /*enabled=*/true));
+}
+
+/**
+ * Initialize the SPI device to be in passthrough mode and allow the following
+ * commands to pass through:
+ *  - ReadJedec
```

Comment:
in the test codes section of the sequence, should we be sending `read jedec` `read sfdp` type commands also? I only remember seeing the normal read commands. It probably doesn't buy you much since this test already proves the connectivity...but just thought i'd ask.

Reply:
Those were not called out as being part of this particular test. However, I filled these out for the heck of it. Originally, I had started writing this code for the bump-in-the-wire type of passthrough setup, with intercepted and uploaded commands.